### ISSUE TITLE
Code for calculating temporal loss.

### ISSUE THREAD

**Issue by Victarry** [daooshee/ReReVST-Code#1]
Hello, thank you for your great work.

I want to know more about how do you implement the calculation of temporal loss for MPI dataset. Could you please share this part of code?

**Comment by daooshee** [comment]
Hope it helps ~
```python
import cv2
import scipy.io as scio

forward_flow = scio.loadmat(flow_list[i], verify_compressed_data_integrity=False)['Img']
mask = cv2.imread(mask_list[i])
mask = 1-mask / 255.

warpped_fake_styled_pre_frame = warp(styled_cur_frame, -forward_flow)
mask *= warp(np.ones([436, 1024,3]), -forward_flow)

warpped_fake_styled_pre_frame = warpped_fake_styled_pre_frame*mask 
styled_pre_frame = styled_pre_frame*mask

temporal_loss = MSE(warpped_fake_styled_pre_frame / 255., styled_pre_frame / 255.) ** 0.5
```

**Comment by Victarry** [comment]
Thank you for your quick reply. ^_^

**Comment by linfengWen98** [comment]
Excuces me, I have one more question.

Given flow predicted by PWC-Net, I would like to know how you generate the mask. Could you please share this part of code?

**Comment by daooshee** [comment]
Hi, linfengWen98
I upload the code to [Baidu Pan](https://pan.baidu.com/s/185pTVUN7RKNSaFMt3xLNWw?pwd=jcai). Sorry that the code does not have comments, and since it's been a long time, I'm also not sure whether the code is correct. So I additionally upload all the generated optical flow and mask files in Baidu Pan (MPI-PWC.zip, 14G). Hope it can solve your problem :)

**Comment by linfengWen98** [comment]
Thank you very much. I get it. :D
